osu.Game/Screens/Play/HUD/GameplayLeaderboard.cs

```diff
@@ -174,6 +174,10 @@ private void sort()
                 orderedByScore[i].ScorePosition = i + 1;
             }
 
+            // change displayed potision to '-' when there are 50 already submitted scores and tracked score is last
+            if (TrackedScore?.ScorePosition == Flow.Count && Flow.Count == 51)
+                TrackedScore.ScorePosition = null;
```

Comment:
This should be based on when the score is the last in the flow, not specifically for a count of "51".

To make this work, you'll also need to differentiate a "solo"/"playlist" leaderboard from a multiplayer one, to handle the case where a multiplayer room always shows all users.

Reply:
The multiplayer part of this comment is valid, but I'm wondering whether checking if the item is last in the flow is 100% correct? I'm thinking about the case where there are 49 or less submitted score for a given beatmap, at which point showing 50 for the play currently in progress would technically not be wrong.

Reply:
Valid point. Checking `>=50` could work if that's referencing a constant that's also sent to the web request somehow.